File: careercup/dynamicproblem/minjumps.go
package main

import (
	"fmt"
	"math"
)

func main()  {
	var input []int = []int{1, 3, 6, 1, 0, 9}
	var len int = len(input)
	jumps := make([]int, len)
	if len == 0 || input[0] == 0 {
		"No jumps"
	}
	jumps[0] = 0
	for i := 1; i < len ;i++  {
		jumps[i] = math.MaxInt64
		for j := 0 ;j < i ;j++  {
			if i <= j + input[j] && jumps[j]  != math.MaxInt64  {
				jumps[i] = int(math.Min( float64(jumps[i]), float64(jumps[j] + 1 )))
			}
		}
	}

	fmt.Println(jumps[len - 1])
}

File: leetcode/src/MinJumps.java
/**
 * Created by chandresh.pancholi on 14/07/16.
 */
public class MinJumps {
    public static void main(String[] args) {
        int[] input = {2, 3, 0, 7};
        int[] jumps = new int[input.length];

        for(int i = 0; i < input.length; i++) {
            jumps[i] = Integer.MAX_VALUE;
        }

        jumps[0] = 0;
        for(int i = 1; i < input.length; i++ ) {
            for(int j = 0; j <i; j++) {
                if(i <= j + input[j] && jumps[j] != Integer.MAX_VALUE) {
                    jumps[i] = Math.min(jumps[i], jumps[j] + 1);
                }
            }
        }
        System.out.println(jumps[input.length - 1]);

    }
}
